test/src/unit-max_tile_size.cc

```diff
@@ -0,0 +1,1237 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests the C API for maximum tile size across all fragments of an array.
+ */
+
+// -d=yes --break "C++ API: Max tile size\, dense\, fixed\, with consolidation",
+
+// -d=yes --break "C++ API: Max fragment size\, sparse\, variable string dimension\, fix attr",
+
+// -d=yes --break "C++ API: Max fragment size\, dense create/evolve, fixed dim/attr",
+
+//  -d=yes --break "C++ API: Max fragment size\, sparse\, variable string dimension\, variable string attribute"
+
+//  -d=yes --break "C++ API: Max fragment size\, sparse\, fix dimension\, variable string attribute", 
+
+//  -d=yes --break "C++ API: Max fragment size\, sparse\, fix dimension\, nullable variable string attribute",
+
+#if TILEDB_SERIALIZATION
+//  -d=yes --break "C++ API: Max tile size\, serialization"
+#endif
+
+
+#include <test/support/tdb_catch.h>
+#include "test/support/src/helpers.h"
+#include "tiledb/common/stdx_string.h"
+#include "tiledb/sm/buffer/buffer.h"
+#include "tiledb/sm/c_api/tiledb_struct_def.h"
+#include "tiledb/sm/cpp_api/tiledb"
+#include "tiledb/sm/cpp_api/tiledb_experimental"
+#include "tiledb/sm/misc/constants.h"
+#include "tiledb/sm/misc/utils.h"
+#include "tiledb/sm/serialization/array.h"
+
+#include <numeric>
+
+struct CAPI_MaxTileSizeFx {
+  //const char* array_name = "cpp_max_tile_size";
+
+  bool showing_data = false;
+
+  CAPI_MaxTileSizeFx()
+    : vfs_(ctx_) {
+  }
+
+  void remove_temp_dir(const std::string& path) {
+    if (vfs_.is_dir(path.c_str())) {
+      vfs_.remove_dir(path.c_str());
+    }
+  };
+
+  template <const int nchars = 1>
+  void create_array_with_attr(const std::string& array_name, const char* attr_name){
+    // Create a TileDB context.
+    // Context ctx;
+    tiledb::Config cfg;
+    tiledb::Context ctx(cfg);
+    tiledb::VFS vfs(ctx);
+
+    remove_temp_dir(array_name.c_str());
+
+    // The array will be 1d with dimension "rows", with domain
+    // [1,4].
+    tiledb::Domain domain(ctx);
+    domain.add_dimension(
+        tiledb::Dimension::create<int>(ctx, "rows", {{1, 4}}, 1));
+
+    // The array will be dense.
+    tiledb::ArraySchema schema(ctx, TILEDB_DENSE);
```

Comment:
Don't like that due to some module's use of both core and cpp APIs which have colliding names.

I prefer the qualification so it's immediately obvious which API is in play rather than having to go figure it out if making changes that may be different depending on which API in play (assumes one is knowledgeable of what at least used to be tiledb::sm::<something> [core usage] vs tiledb::<something> [CPPAPI usage])